### PR TITLE
feat: Add Compaction views and triggers

### PR DESCRIPTION
**Pull request by XuanYang-cn**
- Add Compaction l0 views
- Add Compaction scheduler
- Add Compaction triggerv2
- Add Compaction view manager

See also: #27606

### CODE REVIEW

**Comment by XuanYang-cn** [comment]
rerun ut

**Comment by XuanYang-cn** [comment]
rerun ut

**Review by jiaoew1991 — COMMENTED** [review]
/lgtm

File: internal/datacoord/compaction_scheduler.go [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+	// Start()
+	// Stop()
+	// IsFull() bool
+	// GetCompactionTasksBySignalID(signalID int64) []compactionTask
+}
+
+type CompactionScheduler struct {
+	taskNumber    *atomic.Int32
+	queuingTasks  []*compactionTask
+	parallelTasks map[int64][]*compactionTask
```
[the remark this answers is not in the document]
Comment (jiaoew1991) [reply internal/datacoord/compaction_scheduler.go:27]: You can add it in the next pr.

**Pull request merged**